### PR TITLE
Updating python executable

### PR DESCRIPTION
**Pull request by vsoch**
This PR (to your PR!) will do the following:

- make the python helper an executable (target is environment python) that takes in arguments with argparse (so we explicitly set them)
- uses functions from the python libexec folder. The reason for this is because if this script does get extended to do more, we want to be sure that the functions are (still) tested for python 2 vs. 3, etc.
- (should have) support for python 2 and 3 - there was a call to `.items()` which I think would just be python 3, so I changed that.
- I combined the two loops (one to go through getting defaults and putting in dictionary, and the next going through those to change `0` and `1` to `no` and `yes`, respectively. Was there a reason for having the two? If so, I can change it back.

I gave a quick test to shell into a docker image, and I got this error, I hope it wasn't something that I introduced!

```bash
 sudo singularity shell docker://ubuntu
Cache folder set to /home/vanessa/.singularity/docker
Downloading layer sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4
Extracting /home/vanessa/.singularity/docker/sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4.tar.gz
Extracting /home/vanessa/.singularity/docker/sha256:9332eaf1a55b72fb779d2f249b65855c623c8ce7be83c822b7d80115ef5a3af3.tar.gz
Extracting /home/vanessa/.singularity/docker/sha256:47b5e16c0811b08c1cf3198fa5ac0b920946ac538a0a0030627d19763e2fa212.tar.gz
Extracting /home/vanessa/.singularity/docker/sha256:e931b117db38a05b9d0bbd28ca99a0abe5236a0026d88b3db804f520e59977ec.tar.gz
Extracting /home/vanessa/.singularity/docker/sha256:8f9757b472e7962a4304d4af61630e2cde66129218135b4093a43b9db8942c34.tar.gz
Extracting /home/vanessa/.singularity/docker/sha256:af49a5ceb2a56a8232402f5868cdb13dfdae5d66a62955a73e647e16e9f30a63.tar.gz
/usr/local/libexec/singularity/sexec: symbol lookup error: /usr/local/libexec/singularity/sexec: undefined symbol: singularity_config_init
```

Overall great work! I like how nice and organized this makes the config parsing :O)

### CODE REVIEW

**Comment by vsoch** [comment]
oh hang on, I see a bug

**Comment by vsoch** [comment]
ok :)

File: src/lib/message.c [hunk 1]
```diff
@@ -59,7 +59,7 @@ void _singularity_message(int level, const char *function, const char *file_in,
     const char *file = file_in;
     int syslog_level = LOG_NOTICE;
     char message[512]; // Flawfinder: ignore (messages are truncated to 512 chars)
-    char *prefix = "";
+    char *prefix = NULL;
     va_list args;
```
Comment (vsoch) [inline src/lib/message.c:63]: I rebased with your branch, but I must be missing someone's contributions here.

**Comment by bbockelm** [comment]
There were a few whitespace issues, so I rebased things on the branch.  Accordingly, closing this.

**Comment by vsoch** [comment]
great thanks! Give me a hint about the RPM spec file, and I'll fix it!

**Pull request closed without merging**